packages/lib/src/__tests__/table-generator.test.ts

```diff
@@ -255,4 +259,46 @@ describe('tableGenerator', () => {
 
 		expect(output).toHaveItemValues(['uuid-here', '14.4'])
 	})
+
+	it('throws exception if value missing with no prop', () => {
+		expect(() => tableGenerator.buildTableFromItem(basicData, [{ label: 'Label' }]))
+			.toThrow('both label and value are required if prop is not specified')
+	})
+
+	it('throws exception if label missing with no prop', () => {
+		expect(() => tableGenerator.buildTableFromItem(basicData, [{ value: () => 'some value' }]))
+			.toThrow('both label and value are required if prop is not specified')
+	})
+
+	it('uses empty string for no match', () => {
+		const atSpy = jest.spyOn(_, 'at').mockReturnValue([])
+		const mockLogger = buildMockLogger()
+		const getLoggerSpy = jest.spyOn(logManager, 'getLogger').mockReturnValue(mockLogger)
+
+		const output = tableGenerator.buildTableFromList([{}], ['fieldName'])
+
+		expect(output).toHaveItemValues([''])
+		expect(getLoggerSpy).toHaveBeenCalledTimes(1)
+		expect(getLoggerSpy).toHaveBeenCalledWith('table-manager')
+		expect(atSpy).toHaveBeenCalledTimes(1)
+		expect(atSpy).toHaveBeenCalledWith({}, 'fieldName')
+		expect(debugMock).toHaveBeenCalledTimes(1)
+		expect(debugMock).toHaveBeenCalledWith('did not find match for fieldName in {}')
+	})
+
+	it('combines data on multiple matches', () => {
+		const atSpy = jest.spyOn(_, 'at').mockReturnValue(['one', 'two'])
+		const mockLogger = buildMockLogger()
+		const getLoggerSpy = jest.spyOn(logManager, 'getLogger').mockReturnValue(mockLogger)
+
+		const output = tableGenerator.buildTableFromList([{}], ['fieldName'])
+
+		expect(output).toHaveItemValues(['one, two'])
+		// logger already initialized from previous test
```

Comment:
Ideal unit tests are independent and can run in isolation. This seems to signal we should be clearing some state between these tests.

Reply:
Updated to rebuild tableGenerator every time and added a separate test for this.